lib/core/src/Cardano/Wallet/Primitive/Types/UTxOIndex/Internal.hs

```diff
@@ -473,8 +482,11 @@ indexIsComplete :: UTxOIndex -> Bool
 indexIsComplete u = F.all entryIndexed $ Map.toList $ utxo u
   where
     entryIndexed :: (TxIn, TxOut) -> Bool
-    entryIndexed (i, o) =
-        F.all (indexed i) (TokenBundle.getAssets $ view #tokens o)
+    entryIndexed (i, o)
+        | txOutIsCoin o =
+            i `Set.member` coins u
+        | otherwise =
+            F.all (indexed i) (TokenBundle.getAssets $ view #tokens o)
```

Comment:
👍🏻

Reply:
I do prefer your implementation to [my prototype implementation](https://github.com/input-output-hk/cardano-wallet/compare/jonathanknowles/optimize-ada-only-selection?expand=1), as yours places the index completeness check within one function, which is more concise (and will probably run faster). 👍🏻